Barotrauma/BarotraumaClient/Source/Characters/Character.cs

```diff
@@ -296,17 +296,27 @@ public virtual void DrawFront(SpriteBatch spriteBatch, Camera cam)
 
             if (this == controlled) return;
 
+            //Ideally it shouldn't send the character entirely if we can't see them but /shrug, this isn't the most hacker-proof game atm
+            Limb selfHead = controlled != null ? controlled.AnimController.GetLimb(LimbType.Head) : null;
+            Limb targHead = this.AnimController.GetLimb(LimbType.Head);
+            if (controlled != null && selfHead != null && targHead != null && Submarine.CheckVisibility(selfHead.SimPosition, targHead.SimPosition) != null) //TODO: use Line of Sight instead of CheckVisibility
```

Comment:
Agreed, I'll have to figure out how to only use already calculated LOS checks for visibility instead.

Reply:
Is there a way to detect if any position is within the shadows on the clientside? That kind of function would help a huge deal for this.

Reply:
No IIRC, the LOS effect is just an overlay rendered on top everything but there are no actual calculations done to determine which entities are hidden by it. The way I would do this would be to do the visibility checks every x seconds in the update method and always consider characters outside the screen as invisible.

Reply:
Seems like you're right about the outside-sub though, to be fair CheckVisibility itself will have to be rewritten or something.